packages/common/src/access/can-edit-event.ts

```diff
@@ -0,0 +1,29 @@
+import { IUser } from "@esri/arcgis-rest-types";
+import { IInitiativeModel } from "../types";
+import { getProp } from "../objects";
+import { findBy } from "../util";
+import { hasBasePriv } from "./has-base-priv";
+
+export interface IEventModel {
+  [propName: string]: any;
+  initiative: IInitiativeModel;
+}
+
+/**
+ * Checks if user has access to edit an event in Hub
+ * @param {IEventModel} model consolidated event model as consumed by Hub, contains the event feature, related initiative model, and attendees group
+ * @param {IUser} user
+ * @returns {boolean}
+ */
+export function canEditEvent(model: IEventModel, user?: IUser): boolean {
+  let res = false;
+  if (hasBasePriv(user)) {
+    const coreTeamId = getProp(
+      model,
+      "initiative.item.properties.collaborationGroupId"
+    );
```

Comment:
not sure what is clearer, but i prefer optional chaining -- `model?.initiative?.item?.properties?. collaborationGroupId

Reply:
me too, someone just needs to verify that our build pipeline will properly transpile and/or polyfill them for our target enviornments.

Reply:
looking at our tsconfig, it should be but i haven't verified it. https://www.typescriptlang.org/play?target=2#code/MYewdgzgLgBCBGArGBeGBvGBLAJgLgxngLAFcAbcmAXxoG4AoUSEcgUwDpyQBzACgSIA-B1wj4ASkZA

Reply:
FYI codecov considers each optional bit as a codepath that requires test coverage